CMakeLists.txt

```diff
@@ -125,6 +125,67 @@ option(CUTLASS_SYCL_BUILTIN_ENABLE "Enable this option to use builtin functions
 if (CUTLASS_ENABLE_SYCL)
   set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
+  # Auto-detect IGC version for Intel targets
+  function(detect_igc_version)
+    # Check if IGC_VERSION_MAJOR/MINOR are already set
+    if(IGC_VERSION_MAJOR AND IGC_VERSION_MINOR)
+      # Validate minimum version requirement (IGC 2.18+ for PVC features)
+      if(IGC_VERSION_MAJOR LESS 2 OR (IGC_VERSION_MAJOR EQUAL 2 AND IGC_VERSION_MINOR LESS 18))
+        message(WARNING "User-provided IGC version ${IGC_VERSION_MAJOR}.${IGC_VERSION_MINOR} is below minimum requirement. Enforcing IGC 2.18 minimum for PVC compatibility.")
+        set(IGC_VERSION_MAJOR 2 PARENT_SCOPE)
+        set(IGC_VERSION_MINOR 18 PARENT_SCOPE)
+        message(STATUS "Using enforced minimum IGC version: 2.18")
+      else()
+        message(STATUS "Using user-provided IGC version: ${IGC_VERSION_MAJOR}.${IGC_VERSION_MINOR}")
+      endif()
+      return()
+    endif()
+
+    # Try ocloc query for driver version
+    find_program(OCLOC_EXECUTABLE ocloc)
+    if(OCLOC_EXECUTABLE)
+      execute_process(
+        COMMAND ${OCLOC_EXECUTABLE} query OCL_DRIVER_VERSION
+        OUTPUT_VARIABLE OCL_DRIVER_VERSION
+        ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE
+      )
+      
+      # Extract driver minor version from 25.X.Y format
+      if(OCL_DRIVER_VERSION MATCHES "^25\\.([0-9]+)")
+        set(DRIVER_MINOR ${CMAKE_MATCH_1})
+        
+        # Map driver version to IGC version (enforce minimum 2.18 for PVC compatibility)
```

Comment:
It seems it would be quite a bit easier if we directly do add_compile_defintion of the minor, major of the Neo runtime version and then use that in the code to decide whether the test should be run. Sorry I should have suggested that in the first place. Up to you whether you still want to change this or keep this because it is already implemented.